test/wallet-test.js

```diff
@@ -2279,4 +2299,181 @@ describe('Wallet', function() {
       assert.equal(confirmedCount, 1);
     });
   });
+
+  describe('Wallet Name Claims', function() {
+    // 'it' blocks in this 'describe' create state
+    // that later 'it' blocks depend on.
+    let wallet, update;
+    const network = Network.get('regtest');
+    const workers = new WorkerPool({enabled: false});
+    const wdb = new WalletDB({network, workers});
+    const lockup = 6800503496047;
+    const name = 'cloudflare';
+    const nameHash = rules.hashString(name);
+
+    before(async () => {
+      await wdb.open();
+      wallet = await wdb.create();
+
+      for (let i = 0; i < 3; i++) {
+        const entry = nextEntry(wdb);
+        await wdb.addBlock(entry, []);
+      }
+    });
+
+    after(async () => {
+      await wdb.close();
+    });
+
+    it('should not have any cloudflare state', async () => {
+      const nameinfo = await wallet.getNameState(nameHash);
+      assert.deepEqual(nameinfo, null);
+    });
+
+    it('should confirm cloudflare CLAIM', async () => {
+      // Use a fresh wallet.
+      const pre = await wallet.getBalance();
+      assert.equal(pre.tx, 0);
+      assert.equal(pre.coin, 0);
+      assert.equal(pre.unconfirmed, 0);
+      assert.equal(pre.confirmed, 0);
+      assert.equal(pre.ulocked, 0);
+      assert.equal(pre.clocked, 0);
+
+      const claim = await wallet.sendFakeClaim('cloudflare');
+      assert(claim);
+
+      const tx = claim.toTX(network, wdb.state.height + 1);
+      const entry = nextEntry(wdb);
+      await wdb.addBlock(entry, [tx]);
+
+      const ns = await wallet.getNameState(nameHash);
+      const json = ns.getJSON(wdb.state.height, network);
+      assert.equal(json.name, 'cloudflare');
+      assert.equal(json.state, 'LOCKED');
+
+      const post = await wallet.getBalance();
+      assert.equal(post.tx, 1);
+      assert.equal(post.coin, 1);
+      assert.equal(post.unconfirmed, lockup);
+      assert.equal(post.confirmed, lockup);
+      assert.equal(post.ulocked, lockup);
+      assert.equal(post.clocked, lockup);
+    });
+
+    it('should close the auction', async () => {
```

Comment:
technically, not an auction. maybe "should terminate lockup period" ?

Reply:
"should advance past lockup period?"